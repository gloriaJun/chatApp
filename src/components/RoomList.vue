<template>
  <v-list>
    <v-list-tile
      v-for="item in items"
      :key="item.id"
      avatar
      @click='onClickList(item)'
    >
      <v-list-tile-avatar
        color="indigo"
      >
          <span
            class="white--text headline"
          >
            {{ item.name.charAt(0).toUpperCase() }}
          </span>
      </v-list-tile-avatar>

      <v-list-tile-content>
        <v-list-tile-title v-html="item.name"></v-list-tile-title>
      </v-list-tile-content>
    </v-list-tile>
  </v-list>
</template>

<script>
export default {
  name: 'RoomList',
  props: {
    items: {
      type: [Object, Array],
      default: () => [],
    },
  },
  data: () => ({
  }),
  methods: {
    onClickList(item) {
      this.$emit('click-room', item.id);
    },
  },
};
</script>
