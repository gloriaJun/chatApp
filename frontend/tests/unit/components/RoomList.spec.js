import {
  mount,
} from '@vue/test-utils';
import '../test-setup';

import RoomList from '@/components/RoomList.vue';

describe('RoomList.vue', () => {
  it('snapshot check', () => {
    const wrapper = mount(RoomList);

    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it('snapshot check with items', () => {
    const items = [{
      id: 1,
      name: 'New Year',
    }];
    const wrapper = mount(RoomList, {
      propsData: { items },
    });

    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it('click room', () => {
    const items = [{
      id: 1,
      name: 'New Year',
    }];
    const wrapper = mount(RoomList, {
      propsData: { items },
    });

    wrapper.find('.v-list__tile').trigger('click');
    expect(wrapper.emitted()['click-room'][0]).toEqual([items[0].id]);
  });
});
