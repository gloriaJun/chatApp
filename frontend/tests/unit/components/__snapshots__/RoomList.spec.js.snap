// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomList.vue snapshot check with items 1`] = `
<div
  class="v-list theme--light"
  role="list"
>
  <div
    role="listitem"
  >
    <a
      class="v-list__tile v-list__tile--link v-list__tile--avatar theme--light"
    >
      <div
        class="v-list__tile__avatar"
      >
        <div
          class="v-avatar indigo"
          style="height: 40px; width: 40px;"
        >
          <span
            class="white--text headline"
          >
            
          N
        
          </span>
        </div>
      </div>
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          New Year
        </div>
      </div>
    </a>
  </div>
</div>
`;
